### ISSUE TITLE
Benchmark fails

### ISSUE THREAD

**Issue by tijlleenders** [tijlleenders/ZinZen-scheduler#301]
**Describe the bug**
When attempting to run the benchmark previously developed - it fails.

**To Reproduce**
```
tijl@TFlap:~/ZinZen-scheduler$ cargo run --bin scheduler
   Compiling zinzen v0.1.1 (/home/tijl/ZinZen-scheduler)
    Finished dev [unoptimized + debuginfo] target(s) in 2.05s
     Running `target/debug/scheduler`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected a map", line: 4, column: 16)', src/bin/main.rs:12:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Expected behavior**
Runs the benchmark previously developed.

**Desktop (please complete the following information):**
 - OS: linux

**Additional context**
Tests that need to be adapted are in tests/jsons/benchmark-tests - or just point the main at different test inputs.

**Comment by kobe-reygel** [comment]
I'll look into this

**Comment by kobe-reygel** [comment]
https://github.com/tijlleenders/ZinZen-scheduler/pull/391 should fix flamegraph to run again.

Both binaries (flamegraph and test-summary) now work again.

test-summary does have a bug: rfind only finds **the last** matching string. this means that only the last test module is printed out as a summary.
@tijlleenders does anyone use this binary? I see 2 possible paths
1) we throw away the binary, just running cargo test works just as well. All unnecessary logging has been removed, so the output of cargo test is relevant (apart from the fact that there is an 'OK' line per test, so the output is rather long)
2) I replace the rfind by something more suitable to print **all the summaries** on top of eachother, instead of now only the last module summary

Personally I tend towards option 1). option 2 seems like quite some work for something that is of minor benefit.

**Comment by tijlleenders** [comment]
Agree with option 1.

**Comment by kobe-reygel** [comment]
@tijlleenders option 1 implemented in [e7b2d9f](https://github.com/tijlleenders/ZinZen-scheduler/pull/391/commits/e7b2d9f916f93eb15d7b72b8fa27a30102f5bd94)
